File: examples/node/synth_terminal.js
// A runnable Node.js example: drive a synthetic feed and print a frame.
//
//   ( cd bindings/node && npm install && npm run build )
//   ( cd examples/node && npm install && node synth_terminal.js )

"use strict";

const { Terminal, version } = require("wickra-terminal");

const CONFIG = JSON.stringify({
  sources: [{ Synth: { seed: 1 } }],
  layout: { panels: [{ kind: "Chart", rect: { x: 0, y: 0, w: 100, h: 100 } }] },
});

const term = new Terminal(CONFIG);
term.command(JSON.stringify({ type: "Subscribe", source: 0, symbol: "BTC/USDT" }));

let raw = "";
for (let i = 0; i < 20; i++) {
  raw = term.command(JSON.stringify({ type: "Tick" }));
}

console.log("wickra-terminal", version());
console.log(JSON.stringify(JSON.parse(raw).panels[0], null, 2));

File: examples/python/synth_terminal.py
"""A runnable Python example: drive a synthetic feed and print a frame.

    maturin develop --release -m bindings/python/Cargo.toml
    python examples/python/synth_terminal.py
"""

import json

from wickra_terminal import Terminal

CONFIG = json.dumps(
    {
        "sources": [{"Synth": {"seed": 1}}],
        "layout": {
            "panels": [{"kind": "Chart", "rect": {"x": 0, "y": 0, "w": 100, "h": 100}}]
        },
    }
)


def main() -> None:
    term = Terminal(CONFIG)
    term.command(json.dumps({"type": "Subscribe", "source": 0, "symbol": "BTC/USDT"}))
    raw = ""
    for _ in range(20):
        raw = term.command(json.dumps({"type": "Tick"}))
    print("wickra-terminal", term.version())
    print(json.dumps(json.loads(raw)["panels"][0], indent=2))


if __name__ == "__main__":
    main()
